executor/insert_common.go

```diff
@@ -432,30 +432,39 @@ func (e *InsertValues) fillColValue(datum types.Datum, idx int, column *table.Co
 // For NOT NULL column, it will return error or use zero value based on sql_mode.
 func (e *InsertValues) fillRow(row []types.Datum, hasValue []bool) ([]types.Datum, error) {
 	gIdx := 0
+	gCols := make([]*table.Column, 0, len(e.Table.Cols()))
```

Comment:
maybe we should better NOT pre-alloc this memory(for easy way, maybe we can do it when meet first generated column?), for most user case, the table will be many columns but without any generated column, so this maybe will hurt performance..

Reply:
changed capacity to 0
@lysu  PTAL